File: src/main/java/gay/skitbet/jackiro/model/ServerConfig.java
package gay.skitbet.jackiro.model;

import gay.skitbet.mongoy.annotation.CollectionName;
import gay.skitbet.mongoy.annotation.IdField;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

@Data
@Accessors(fluent = true)
@ToString
@CollectionName("servers_name")
public class ServerConfig {

    @IdField
    public String guildId;

    public String botUpdateChannelId;
    public ArrayList<String> disabledModules = new ArrayList<>();

    public List<String> disabledCommands = new ArrayList<>();
    public HashMap<String, ServerUserData> userData = new HashMap<>();

    // XP Settings
    public boolean xpEnabled = true;
    public double xpMultiplier = 1.0;
    public int xpPerMessage = 10;
    public int cooldownSeconds = 60;
    public int baseXpToLevelUp = 100;

    public ServerConfig(String guildId) {
        this.guildId = guildId;
    }

    public ServerConfig() {}
}

File: src/main/kotlin/gay/skitbet/jackiro/model/ServerConfig.kt
package gay.skitbet.jackiro.model

import gay.skitbet.mongoy.annotation.CollectionName
import gay.skitbet.mongoy.annotation.IdField

@CollectionName("servers_name")
data class ServerConfig(
    @IdField
    var guildId: String = "",

    var botUpdateChannelId: String? = null,
    var disabledModules: ArrayList<String> = ArrayList(),
    var disabledCommands: MutableList<String> = mutableListOf<String>(),
    var userData: HashMap<String, ServerUserData> = HashMap(),

    // XP Settings
    var xpEnabled: Boolean = true,
    var xpMultiplier: Double = 1.0,
    var xpPerMessage: Int = 10,
    var cooldownSeconds: Int = 60,
    var baseXpToLevelUp: Int = 100
)
